### ISSUE TITLE
Analyzer should detect object class after assertInstanceOf call

### ISSUE THREAD

**Issue by ossinkine** [phpstan/phpstan-phpunit#10]
Let's consider these example
```php
class Foo
{
    public function getBar()
    {
    }
}

class FooTest extends \PHPUnit\Framework\TestCase
{
    public function testFoo()
    {
        $foo = $this->getFoo();
        $this->assertInstanceOf(Foo::class, $foo);
        $this->assertNull($foo->getBar());
    }

    /**
     * @return object
     */
    private function getFoo()
    {
        // ...
    }
}
```
PHPStan alerts `Call to an undefined method object::getBar().` but we can be sure that `$foo` is object of `Foo` class after `assertInstanceOf` call.
`/** @var Foo $foo */` can be used as workaround.

**Comment by ondrejmirtes** [comment]
Hi, this is a duplicate of https://github.com/phpstan/phpstan-phpunit/issues/9 (previous issue), see the discussion there.
